cli/command/service/rollback.go

```diff
@@ -56,8 +57,7 @@ func runRollback(dockerCli command.Cli, flags *pflag.FlagSet, options *serviceOp
 
 	fmt.Fprintf(dockerCli.Out(), "%s\n", serviceID)
 
-	if options.detach {
-		warnDetachDefault(dockerCli.Err(), apiClient.ClientVersion(), flags, "rolled back")
+	if options.detach || versions.LessThan(apiClient.ClientVersion(), "1.29") {
```

Comment:
One thing I was thinking is if we should put the version-check somewhere central, and just change `options.detach` to `true`, based on that version; this would prevent possible bugs if more commands get an interactive variant.

Reply:
I don't think we need the version check anymore. The behaviour is client-side only, right? So even if they are running against an older daemon we should keep the same default for detach. I think we can remove the version check.

Reply:
I wasn't sure if older daemons would return the right response for the `waitOnService()` to work correctly; if that's not an issue then I am +1 on removing

Reply:
Apparently it depends on this `_up-to-date=true` filter which was added in 17.05, so I guess it does depend on this version still.